html/beps/bep_encrypted_data.rst

```diff
@@ -0,0 +1,297 @@
+
+Metadata format
+===============
+
+
+.. parsed-literal::
+
+    {
+      info: {
+        bepXX: {
+          mac: *<32bytes of hmac output (string)>*,
+          salt: *<32bytes of random binary data (string)>*,
+          shadow: *<optional, bencoded-then-encrypted dictionary (string)>*
+          v: *<version (integer)>*,
+        },
+        length *or* files: *<unchanged>*,
+        name: *<public name (string)>*,
+        piece length: *<unchanged>*
+        pieces: *<N*20 bytes, piece hashes of the payload ciphertext (string)>*
+      },
+    }
+
+
+``salt``
+  the random data must be generated by a cryptographically secure RNG to avoid IV reuse.
+
+``v``
+  The protocol version used to encrypt the torrent, currently *1*. New versions may be introduced by updates to this BEP if cryptographic weaknesses necessitate incompatible changes. Implementations should check if they support the version indicated in the metadata file and otherwise inform the user that they can download the data but not decrypt it.
+
+``shadow``
+  bencoded-then-encrypted dictionary of key-value pairs that shadow entries in the info dictionary.
+  If it is absent only the payload is encrypted and all info dictionary entries are non-shadowed.
+  Implementations should only shadow a whitelist of keys which they have a shadowing strategy and ignore other keys.
+  Shadowable keys suggested by this BEP: ``length, files, name, comment``.
+
+``mac``
+  message authentication code covering the info dictionary
+
+``name``
+  the name field is a mandatory part of [BEP 3]_, therefore a placeholder MUST be provided if a shadow name is used. An implementation may either generate a random string consisting of filesystem-friendly characters or allow the user to choose a public name that reveals less information than the shadow name.
+
+``files``, ``length``
+  The shadow dictionary MAY override the single/multifile nature indicated by the public info dictionary. If it does not shadow either key then the public information is canonical.
+  If the ``files`` or ``length`` are shadowed then the overall payload length MUST be consistent with the public version.
+  If a shadow dictionary is present the public information should be treated as decorative / advisory until it can be determined whether it has been shadowed, i.e. until the shadow data can be decrypted. 
+
+
+To protect privacy an implementation should use shadowing for any additional keys that reveal information about the payload
+
+
+Encryption
+==========
+
+Building blocks used:  SHA2-256[rfc6234]_, ChaCha20[rfc7539]_, HMAC[rfc2104]_, PBKDF2[rfc2898]_  
```

Comment:
maybe it would make sense to be explicit about that these are the primitives for version 1 (i.e. `v` = `1`).